src/coredump/coredump.c

```diff
@@ -1212,8 +1218,25 @@ static int process_kernel(int argc, char* argv[]) {
 
         if (context.is_journald || context.is_pid1)
                 r = submit_coredump(&context, iovw, STDIN_FILENO);
-        else
+        else {
+                _cleanup_free_ char *process_name;
+                process_name = NULL;
+                process_name = filename_escape(context.meta[META_COMM]);
+                if (!process_name)
+                        log_error_errno(r, "Failed to escape process name. Ignoring coredump ratelimit: %m");
+                else {
+                        /* per process coredump ratelimit */
+                        r = coredump_ratelimit(process_name, arg_ratelimitinterval, arg_ratelimitburst);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to process coredump ratelimit: %m");
```

Comment:
duplcate logging.

generally we require functions to either log about all their errors or about none (in which case the caller should do that). That way we can be sure we don't get duplicate logging or missing logging. Here your funciton logs about some errors but not on others, and thus you get logging once or twice, depending on the condition.

Reply:
also, you ignore the failure here mostly, and just continue. This usually means you should log at log_warning() level at most, not log_error()

Reply:
I have addressed the review comment. Please check once